contracts/InsuranceArbitrator.sol

```diff
@@ -59,34 +72,31 @@ contract InsuranceArbitrator {
         address indexed insurer,
         string insuredEvent,
         address indexed insuredAddress,
-        address currency,
+        IERC20 currency,
         uint256 insuredAmount
     );
     event ClaimSubmitted(
         uint256 claimTimestamp,
         bytes32 indexed policyId,
-        address indexed insurer,
```

Comment:
I now realize that `insurer` is not part of `InsurancePolicy` struct so we cannot retrieve it for logs. If this is required for additional functionality we can add it back later.